src/tribler-gui/tribler_gui/event_request_manager.py

```diff
@@ -131,6 +129,7 @@ def on_finished(self):
         self.connect_timer.setSingleShot(True)
         self.connect_timer.timeout.connect(self.connect)
         self.connect_timer.start(500)
+        self.tribler_started_flag = False
```

Comment:
**issue:** It is a bit dangerous to unset the flag here because it can lead to a race condition. If `tribler_started_flag` is set to True and then back to False quickly, it might remain unnoticed, as if it was never set to True.

**suggestion:** Probably, it is not necessary to set the flag back to False. Or you can introduce a separate `tribler_stopped_flag` and set it to True here.

**thought:**  The way we process Qt signals may be unsafe in some places: if a signal is raised too quickly, a slot might be not attached to the signal at this moment yet. Ideally, we should have a corresponding asyncio Event for each one-time PyQt signal, and the corresponding slot should just set this event. Then the Tribler code can wait for this event, and it will work correctly without a race condition even if the signal was triggered before we start waiting for the event.

Reply:
Good point!
The problem is, we don't use `asyncio` in the GUI. The reactor is run by QT, and there is no such thing as a "single-shot event" in QT. So, we have to invent something of our own.
I'll try to make this a bit safer :+1:

Reply:
Also, the follow-up to this PR will remove the `TRIBLER_STARTED` and the event entirely :wink: